packages/typespec-vscode/src/telemetry/telemetry-client.ts

```diff
@@ -158,6 +159,39 @@ export class TelemetryClient {
     }
   }
 
+  public async doOperationWithOperationDetailTelemetry<T>(
```

Comment:
i think you just need to call logOperationDetailTelemetry before and after compile to log the time, isn't it?

Reply:
@RodgeFu  We can record down the start time and then call logOperationDetailTelemetry at every branch of the compile operation (success or fail).
I just wrap all those behavior in one function, so in the place where we need log such telemetry, we just need call this function.

in the future, we may need to record emitStartTime/emitEndTime for each emitting. So I wrap it for convenience.

Reply:
i mean you just need to call the telemetryclient.logOperaitonDetailTelemetry with the proper activity id which is enough. it will send a detail telemetry record which can be associated with the operation with the activity_id.

Reply:
> i mean you just need to call the telemetryclient.logOperaitonDetailTelemetry with the proper activity id which is enough. it will send a detail telemetry record which can be associated with the operation with the activity_id.

Do you want to send two events, one to record startTime, the other to record endTime? I am considering we can record startTime and endTime in one event, so that the number of events will reduce.

Reply:
change to send two events, one to record startTime, the other to record endTime. Then we don't need to wrap a function . remove it.